rasa/nlu/training_data/formats/rasa_yaml.py

```diff
@@ -322,3 +350,108 @@ def is_yaml_nlu_file(filename: Text) -> bool:
                 f"Error: {e}"
             )
         return False
+
+
+class RasaYAMLWriter(TrainingDataWriter):
+    """Writes training data into a file in a YAML format."""
+
+    def dumps(self, training_data: "TrainingData") -> Text:
+        """Turns TrainingData into a string."""
+        stream = StringIO()
+        self.dump(stream, training_data)
+        return stream.getvalue()
+
+    def dump(self, filename: Text, training_data: "TrainingData") -> None:
+        """Writes training data into a file in a YAML format.
+
+        Args:
+            filename: Name of the target file.
+            training_data: TrainingData object.
+        """
+        from rasa.validator import KEY_TRAINING_DATA_FORMAT_VERSION
+        from ruamel.yaml.scalarstring import DoubleQuotedScalarString
+
+        nlu_items = []
+        nlu_items.extend(self.process_intents(training_data))
+        nlu_items.extend(self.process_synonyms(training_data))
+        nlu_items.extend(self.process_regexes(training_data))
+        nlu_items.extend(self.process_lookup_tables(training_data))
+
+        result = OrderedDict()
+        result[KEY_TRAINING_DATA_FORMAT_VERSION] = DoubleQuotedScalarString(
+            LATEST_TRAINING_DATA_FORMAT_VERSION
+        )
+        result[KEY_NLU] = nlu_items
+
+        io_utils.write_yaml(result, filename, True)
+
+    @staticmethod
```

Comment:
should this rather be a classmethod, e.g. we could call`cls.process_training_examples_by_key`

Reply:
makes sense.